src/js/client/editprefset-viewport.js

```diff
@@ -0,0 +1,681 @@
+"use strict";
+
+var gpii  = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.devpmt");
+
+/**
+ * Edit Prefs is the main component for driving the page editing
+ * a preference set.
+ */
+fluid.defaults("gpii.devpmt.editPrefs", {
+    gradeNames: ["gpii.handlebars.templateAware", "gpii.binder.bindOnCreate", "gpii.binder.bindOnDomChange"],
+    mergePolicy: {
+        allSolutions: "noexpand",
+        commonTerms: "noexpand"
+    },
+    model: {
+        flatPrefs: {},
+        commonTerms: [],
+        commonTermsSorted: [],
+        contextNames: [],
+        npsetApplications: [],
+        allSolutions: {},
+        unsavedChangesExist: false,
+        unsavedChanges: [],
+
+        // Generic Prefs Filters
+        settingsFilter: "mysettings",
+        settingsSearch: "",
+
+        // A hash which could include an entry for each
+        // settingsTables dynamicComponent that contains
+        // it's local display options (such as mine/all settings),
+        // kept here in case we need to rerender the entire page
+        // for a major prefset change.
+        productTableFilters: {},
+
+        devModeOn: false,
+
+        // Information for if we are currently editing a setting,
+        // and if so, which context/term/product it's for and a
+        // copy of the metadata.
+        currentlyEditing: {
+            active: false,
+            current: {
+                context: "",
+                term: "",
+                value: "",
+                product: ""
+            },
+            metadata: {
+                name: "",
+                description: "",
+                schema: {}
+            }
+        },
+
+        commonTermUsageCounts: {},
+        productTermUsageCounts: [],
+
+        // Contains any free information that pertains to an active dialog.
+        // While we can only display one modal dialog at a time, the options
+        // and data for a dialog can change between instances, such as adding
+        // different products to an NP Set, and presenting an confirm dialog
+        // each time.
+        //
+        // notes: I had initially wanted to do this, by passing options to the
+        // createOnEvent firing to create the deffered component, but it didn't
```

Comment:
Presumably you have run into https://issues.fluidproject.org/browse/FLUID-5912 - the workaround is to write the initial binding onto the fire arguments in the plain options area

Reply:
this workaround is completely removed now